charts/simple-app/values.yaml

```diff
@@ -122,6 +122,46 @@ readinessProbe:
     path: /
     port: http
 
+# An optional init container. This container starts and exits before the
```

Comment:
So... I definitely don't want to create a dedicated configuration here for the initContainer. I would consier a single large parameter (list of maps)) that _is_ the `initContainers` value. The thing is that once you have one initContainer, who is to say you won't have two, or three... there's just too much customization available at that point.

I'd think:

```
# values.yaml
initContainers: []
```

```
# deployment.yaml
apiVersion: v1
...
spec:
  template:
    {{- with .Values.initContainers }}
    {{- tpl (toYaml .) $ | nindent XX }}
    {{- end }}
...
```

Even this though you should prototype on your laptop in your own chart and just see how it feels

Reply:
interesting.  as implemented, this has one bespoke initContainer, whose configuration is limited by the bits exposed in `values.yaml`.  This sounds like you would prefer a blind import from `values.yaml` with unlimited configuration.  Blindly passing `initContainers` through to the pod spec is certainly more powerful, but it is also a lot more rope for the user to hang themselves with.

Reply:
yup... I feel like the moment we're talking about something as complex and powerful as an initContainer, we should release the reigns.

Reply:
ok.  see #173 for that approach.